src/libsyntax/parse/literal.rs

```diff
@@ -134,9 +134,9 @@ impl LitKind {
         let (kind, symbol, suffix) = match *self {
             LitKind::Str(symbol, ast::StrStyle::Cooked) => {
                 // Don't re-intern unless the escaped string is different.
-                let s: &str = &symbol.as_str();
+                let s = symbol.as_str();
                 let escaped = s.escape_default().to_string();
```

Comment:
This feels like it could avoid allocating until a need to escape is found.
(not relevant to the PR, just spotted it, likely not relevant to perf)

Reply:
Profiles show that it's not at all hot for any of the rustc-perf benchmarks.